ui/src/message_popup/experience_page.jsx

```diff
@@ -117,24 +98,32 @@ export default React.createClass({
     this.setState(this.getInitialState());
   },
 
+  onSaveScaffoldingAndSession(scaffolding, gameSession){
+    this.setState({
+      scaffolding: {
```

Comment:
You can simplify the syntax a bit, since
```
this.setState({
  scaffolding: {...scaffolding}
})
```

is equivalent to 

```
this.setState({
  scaffolding
});
```